src/main/java/net/dries007/tfc/compat/wailatop/WailaBlocks.java

```diff
@@ -0,0 +1,502 @@
+            }
+        }
+        currenttip.add(key);
+
+        return currenttip;
+    }
+
+    public List<String> OreTFCBody(ItemStack itemStack, List<java.lang.String> currenttip, IWailaDataAccessor accessor, IWailaConfigHandler config)
+    {
+
+        BlockOreTFC b = (BlockOreTFC) accessor.getBlock();
+        int metadata = accessor.getMetadata();
+        Ore.Grade gradevalue = Ore.Grade.valueOf(metadata);
+        Metal metal = b.ore.getMetal();
+        String orename = b.ore.toString();
+        String key;
+
+        if (gradevalue == Ore.Grade.NORMAL)
+        {
+            key = new TextComponentTranslation("waila.normal").getFormattedText() + " " + new TextComponentTranslation("item.tfc.ore." + orename + ".name").getFormattedText();
+        }
+        else
+        {
+            String gradename = gradevalue.toString().toLowerCase();
+            key = new TextComponentTranslation("item.tfc.ore." + orename + "." + gradename + ".name").getFormattedText();
+        }
+        currenttip.add(key);
+        if (metal != null && config.getConfig("tfc.newtotfc"))
+        {
+            currenttip.add("(" + new TextComponentTranslation(metal.getTranslationKey()).getFormattedText() + ")");
+        }
+
+        return currenttip;
+    }
+
+    public ItemStack OreTFCStack(IWailaDataAccessor accessor, IWailaConfigHandler config)
+    {
+
+        BlockOreTFC b = (BlockOreTFC) accessor.getBlock();
+        BlockStateContainer state = b.getBlockState();
+        ItemStack itemstack = ItemOreTFC.get(b.ore, 1);
+
+        return itemstack;
+    }
+
+    private List<String> BlastFurnaceBody(ItemStack itemStack, List<String> currenttip, IWailaDataAccessor accessor, IWailaConfigHandler config)
+    {
+        //todo
+        TEBlastFurnace te = (TEBlastFurnace) accessor.getTileEntity();
+        NBTTagCompound nbt = accessor.getNBTData();
+        // Something is borked with Waila reading the nbt second time round. Dunno
+        return currenttip;
+    }
+
+    private List<String> FruitTreeHead(ItemStack itemStack, List<String> currenttip, IWailaDataAccessor accessor, IWailaConfigHandler config)
+    {
+        String name = accessor.getBlock().getTranslationKey();
+        String output = "waila.fruit_trees." + name.substring(name.lastIndexOf(".") + 1) + ".name";
+
+        currenttip.set(0, new TextComponentTranslation(output).getFormattedText());
+
+        return currenttip;
+    }
+
+    private ItemStack CropSimpleStack(IWailaDataAccessor accessor, IWailaConfigHandler config)
+    {
+
+        BlockCropTFC b = (BlockCropTFC) accessor.getBlock();
+        ICrop crop = b.getCrop();
+        IBlockState blockstate = accessor.getBlockState();
+        Integer curStage = blockstate.getValue(b.getStageProperty());
+        ItemStack foodDrop = crop.getFoodDrop(curStage);
+
+
```

Comment:
```suggestion
        BlockCropTFC b = (BlockCropTFC) accessor.getBlock();
        ICrop crop = b.getCrop();
        IBlockState blockstate = accessor.getBlockState();
        int curStage = blockstate.getValue(b.getStageProperty());
        ItemStack foodDrop = crop.getFoodDrop(curStage);
```
Some odd whitespace, and boxed primitives.